src/main/plugin/iso19115-3/formatter/groovy/iso191153/Handlers.groovy

```diff
@@ -0,0 +1,465 @@
+        handlers.add name: 'Container Elements', select: matchers.isContainerEl, priority: -1, commonHandlers.entryEl(f.&nodeLabel, addPackageViewClass)
+
+        commonHandlers.addDefaultStartAndEndHandlers();
+        addExtentHandlers()
+
+        handlers.sort name: 'Text Elements', select: matchers.isContainerEl, priority: -1, sortContainerEl
+    }
+
+    def sortContainerEl = {el1, el2 ->
+        def v1 = matchers.isContainerEl(el1) ? 1 : -1;
+        def v2 = matchers.isContainerEl(el2) ? 1 : -1;
+        return v1 - v2
+    }
+    def addPackageViewClass = {el -> if (packageViews.contains(el.name())) return el.name().replace(':', '_')}
+
+    def addExtentHandlers() {
+        handlers.add commonHandlers.matchers.hasChild('gex:EX_Extent'), commonHandlers.flattenedEntryEl({it.'gex:EX_Extent'}, f.&nodeLabel)
+        handlers.add name: 'BBox Element', select: matchers.isBBox, bboxEl(false)
+        handlers.add name: 'Polygon Element', select: matchers.isPolygon, polygonEl(false)
+        handlers.add 'gex:geographicElement', commonHandlers.processChildren{it.children()}
+        handlers.add 'gmd:extentTypeCode', extentTypeCodeEl
+    }
+
+    def isoTextEl = { isofunc.isoTextEl(it, isofunc.isoText(it))}
+    def isoUrlEl = { isofunc.isoUrlEl(it, isofunc.isoUrlText(it), isofunc.isoUrlText(it))}
+    def isoAnchorUrlEl = { isofunc.isoUrlEl(it, isofunc.isoAnchorUrlLink(it), isofunc.isoAnchorUrlText(it))}
+    def isoDatasetUriEl = { isofunc.isoUrlEl(it, isofunc.isoText(it), isofunc.isoText(it))}
+    def isoCodeListEl = {isofunc.isoTextEl(it, f.codelistValueLabel(it))}
+    def isoBasicType = {isofunc.isoTextEl(it, it.'*'.text())}
+    def isoSimpleTextEl = { isofunc.isoTextEl(it, it.text()) }
+    def isoSimpleTextElGrouped = { elems ->
+        def listItems = elems.findAll{!it.text().isEmpty()}.collect {it.text()};
+        handlers.fileResult("html/list-entry.html", [label:f.nodeLabel(elems[0]), listItems: listItems])
+    }
+    def parseBool(text) {
+        switch (text.trim().toLowerCase()){
+            case "1":
+            case "true":
+            case "y":
+                return true;
+            default:
+                return false;
+        }
+    }
+    def isoBooleanEl = {isofunc.isoTextEl(it, parseBool(it.'*'.text()).toString())}
+    def dateEl = {isofunc.isoTextEl(it, isofunc.dateText(it));}
+    def extentTypeCodeEl = {
+        isofunc.isoTextEl(it, parseBool(it.text()) ? 'include' : 'excluded')
+    }
+    def ciDateEl = {
+        if(matchers.isDateEl(it.'cit:date')) {
+            def dateType = f.codelistValueLabel(it.'cit:dateType'.'cit:CI_DateTypeCode')
+            commonHandlers.func.textEl(dateType, isofunc.dateText(it.'cit:date'));
+        }
+    }
+    def localeEls = { els ->
+        def locales = []
+        els.each {
+            it.'gmd:PT_Locale'.each { loc ->
```

Comment:
lan here, and not gmd after for charset and language. Charset is now characterEncoding